Directory.Build.props

```diff
@@ -0,0 +1,47 @@
+<Project>
+
+  <PropertyGroup>
+    <LangVersion>preview</LangVersion>
+    <EnableNETAnalyzers>true</EnableNETAnalyzers>
+    <AnalysisLevel>preview</AnalysisLevel>
+    <NoWarn>$(NoWarn);CS1591;NU5118;NU5128</NoWarn>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>portable</DebugType>
+    <SymbolPackageFormat>snupkg</SymbolPackageFormat>
+    <CodeAnalysisRuleset>$(MSBuildThisFileDirectory)CodeAnalysis.ruleset</CodeAnalysisRuleset>
+    <EnableXlfLocalization>false</EnableXlfLocalization>
```

Comment:
There are some properties here that I am not familiar with. Have you introduced them on purpose, or are they generated by a tool, without you having any special thought about them? I'm thinking of:

* EnableXlfLocalization
* AccelerateBuildsInVisualStudio

Do these make sense if we don't publish grate as a traditional NuGet library (e.g. for people to reference in their code)?

* ProduceReferenceAssembly
* SymbolPackageFormat